compare.go

```diff
@@ -72,6 +72,61 @@ func platformVector(platform specs.Platform) []specs.Platform {
 		if variant == "" {
 			variant = "v8"
 		}
+
+		if armMajor, err := strconv.Atoi(strings.TrimPrefix(variant[:2], "v")); err == nil && armMajor >= 8 {
+			armMinor := 0
+			if len(variant) == 4 {
+				if minor, err := strconv.Atoi(variant[3:]); err == nil && variant[2] == '.' {
+					armMinor = minor
+				}
+			}
+
+			if armMajor == 9 {
+				for minor := armMinor - 1; minor >= 0; minor-- {
+					arm64Variant := "v" + strconv.Itoa(armMajor) + "." + strconv.Itoa(minor)
+					if minor == 0 {
+						arm64Variant = "v" + strconv.Itoa(armMajor)
+					}
+					vector = append(vector, specs.Platform{
+						Architecture: platform.Architecture,
+						OS:           platform.OS,
+						OSVersion:    platform.OSVersion,
+						OSFeatures:   platform.OSFeatures,
+						Variant:      arm64Variant,
+					})
+				}
+
+				// v9.0 diverged from v8.5, meaning that v9.x is compatible with v8.{x+5}
```

Comment:
I keep linking to https://github.com/golang/go/issues/60905#issuecomment-1611867608 to describe this, but I imagine (vain hope, perhaps?) there's probably a better / more "canonical" reference for this complexity? :sob:

Reply:
It's mentioned in https://github.com/golang/go/issues/60905#issuecomment-1611815533:

From [ARM Architecture Reference Manual](https://developer.arm.com/documentation/ddi0487/latest/), section A2.3 describes "Armv9-A architecture extensions", and then its subsections has the following:

- A2.3.1 The Armv9.0 architecture extension
  - It is Armv8.5 compliant.
- A2.3.2 The Armv9.1 architecture extension
  - It is Armv8.6 compliant.
  - It is Armv9.0 compliant.
- A2.3.3 The Armv9.2 architecture extension
  - It is Armv8.7 compliant.
  - It is Armv9.1 compliant.
- A2.3.4 The Armv9.3 architecture extension
  - It is Armv8.8 compliant.
  - It is Armv9.2 compliant.
- A2.3.5 The Armv9.4 architecture extension
  - It is Armv8.9 compliant.
  - It is Armv9.3 compliant.

Looks like there's no Armv9.5 yet

Reply:
Announced by not equivalent 8.10 or 8.11 for 9.6?